### PR TITLE
Added test cases relate to Network Control Module

### PR DESCRIPTION
**Pull request by sandeep-vedam**
Following are the test cases that are added in this PR : 
NetworkControl_Assign_001.test.js
NetworkControl_Assign_002.test.js
NetworkControl_Flush_001.test.js
NetworkControl_Flush_002.test.js
NetworkControl_Network_001.test.js
NetworkControl_Network_002.test.js
NetworkControl_Reload_001.test.js
NetworkControl_Reload_002.test.js
NetworkControl_Request_001.test.js
NetworkControl_Request_002.test.js
NetworkControl_Status_001.test.js
NetworkControl_Status_002.test.js
NetworkControl_Status_003.test.js

### CODE REVIEW

File: src/tests/networkcontrol/NetworkControl_Assign_001.test.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      assert: 'activated',
+    },
+    {
+      description: 'Invoke Assign',
+      sleep: 5,
+      test() {
+        return assignNetworkAdapter().call(this, 'eth0')
+      },
+      validate(res) {
+        //TODO - Update validation by confirming that assign happened
```
Comment (wouterlucas) [inline src/tests/networkcontrol/NetworkControl_Assign_001.test.js:31]: How can we actually validate that the test has happened? Just be checking if the device is still online?
@pwielders @MFransen69  what does the "assign" do on the network adapter?

File: src/tests/networkcontrol/NetworkControl_Flush_001.test.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      assert: 'activated',
+    },
+    {
+      description: 'Invoke Flush',
+      sleep: 5,
+      test() {
+        return flushNetworkAdapter.call(this, 'eth0')
+      },
+      validate(res) {
+        //TODO - Update validation by confirming that flush happened
```
Comment (wouterlucas) [inline src/tests/networkcontrol/NetworkControl_Flush_001.test.js:31]: Same here, can we only check if the device is still reachable?

File: src/tests/networkcontrol/NetworkControl_Assign_001.test.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      test: pluginDeactivate,
+      params: constants.networkControlPlugin,
+      assert: 'deactivated',
+    },
+    {
+      description: 'Check if NetworkControl Plugin is started correctly',
+      test: pluginActivate,
+      params: constants.networkControlPlugin,
+      assert: 'activated',
+    },
```
Comment (wouterlucas) [inline src/tests/networkcontrol/NetworkControl_Assign_001.test.js:23]: Should we really be starting/restating the network plugin prior to each test?

Or could we just test the starting/stopping of the plugin once and run other flush/assign/etc tests without restarting the plugin?

File: src/tests/networkcontrol/NetworkControl_Reload_001.test.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      assert: 'activated',
+    },
+    {
+      description: 'Invoke Reload',
+      sleep: 5,
+      test() {
+        return reloadNetworkAdapter.call(this, 'eth0')
+      },
+      validate(res) {
+        //TODO - Update validation by confirming that reload happened
```
Comment (wouterlucas) [inline src/tests/networkcontrol/NetworkControl_Reload_001.test.js:31]: Same here, should we just check if the device is still reachable? Assuming it did not get another IP :)

File: src/tests/networkcontrol/NetworkControl_Request_001.test.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+      assert: 'activated',
+    },
+    {
+      description: 'Invoke Request',
+      sleep: 5,
+      test() {
+        return requestNetworkAdapter.call(this, 'eth0')
+      },
+      validate(res) {
+        //TODO - Update validation by confirming that Request happened
```
Comment (wouterlucas) [inline src/tests/networkcontrol/NetworkControl_Request_001.test.js:31]: ditto

File: src/tests/networkcontrol/NetworkControl_Status_001.test.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+      params: constants.networkControlPlugin,
+      assert: 'activated',
+    },
+    {
+      description: 'Invoke Up to check network status',
+      sleep: 5,
+      test() {
+        return getNetworkStatus.call(this, 'eth0')
+      },
+      validate(res) {
```
Comment (wouterlucas) [inline src/tests/networkcontrol/NetworkControl_Status_001.test.js:30]: probably check if it still online after doing the up?

File: src/tests/networkcontrol/NetworkControl_Status_003.test.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+      description: 'Check if NetworkControl Plugin is started correctly',
+      test: pluginActivate,
+      params: constants.networkControlPlugin,
+      assert: 'activated',
+    },
+    {
+      description: 'Invoke Up to check network status',
+      sleep: 5,
+      test() {
+        return setNetworkStatus.call(this, 'eth0', 'false') //TODO - Network status cannot be set to false as the connection is lost.
```
Comment (wouterlucas) [inline src/tests/networkcontrol/NetworkControl_Status_003.test.js:28]: We could support this if we add user interaction?

E.g. we could:
- Warn the user his box is going down after running this test
- Execute eth down
- Test if the box is really down (fail if its still online)
- Prompt user to reboot it

wdyt @michielvandergeest @Divya-563 ?

File: src/tests/networkcontrol/NetworkControl_Assign_001.test.js (hunk 1 above)
Comment (sandeep-vedam) [reply src/tests/networkcontrol/NetworkControl_Assign_001.test.js:31]: Hi @wouterlucas ,

The tests were updated with TODO, as I don't know the validation for these and asked Divya how to validate assign/ flush / reload / request (Sent attachment in the Email with my queries). Once I get clarity on these I will update all of these.

File: src/tests/networkcontrol/NetworkControl_Assign_001.test.js (hunk 3 above)
Comment (sandeep-vedam) [reply src/tests/networkcontrol/NetworkControl_Assign_001.test.js:23]: This is to make sure the test is independent and no previous test run affects this test. Please let me know if you have better idea to achieve this.

File: src/tests/networkcontrol/NetworkControl_Assign_001.test.js (hunk 1 above)
Comment (wouterlucas) [reply src/tests/networkcontrol/NetworkControl_Assign_001.test.js:31]: Lets park the assign tests for now so we can move the networkcontrol tests forward.

**Pull request closed without merging**